### PR TITLE
Ensure query/scan returned cursor is undefined when there are no more records to scan/query

### PR DESCRIPTION
**Pull request by teeeteee**

### CODE REVIEW

File: src/test/dynamo/query.test.ts [hunk 1]
```diff
@@ -230,6 +239,33 @@ async function testQueryWithLimit(jayZ?: JayZ) {
...
+  const { value: response2 } = await db
+    .query(MusicianPartition.key({ id: musician.id }))
+    .iterator({
+      cursor: response1.cursor,
+      pageSize: 1
+    })
+    .next()
+
+  expect(response2.items).toEqual({ musician: [], song: [] })
+  expect(response2.cursor).toBeUndefined()
```
Comment (gbryan) [inline src/test/dynamo/query.test.ts:266]: Nice!

File: src/test/dynamo/scan.test.ts [hunk 2]
```diff
@@ -246,3 +255,26 @@ async function testScanWithLimit(jayZ?: JayZ) {
...
+
+  const { value: response2 } = await db
+    .scan()
+    .iterator({
+      cursor: response1.cursor,
+      pageSize: 1
+    })
+    .next()
+
+  expect(response2.items).toEqual({ musician: [], song: [] })
```
Comment (gbryan) [inline src/test/dynamo/scan.test.ts:279]: Let's also make sure the cursor is `undefined` here: `expect(response2.cursor).toBeUndefined()`.

**Review by gbryan — COMMENTED** [review]
Great catch, Titi! Just requesting a small update to the 2nd test.

File: src/test/dynamo/scan.test.ts (hunk 2 above)
Comment (teeeteee) [reply src/test/dynamo/scan.test.ts:279]: Ah yea, good catch

**Review by gbryan — APPROVED** [review]
Thanks!

**Review by jcarver989 — APPROVED** [review]
Thanks Titi!

I somehow totally missed this PR and ended up opening one that had the same fix: https://github.com/ginger-io/beyonce/pull/56. But you actually have more robust tests here, so thank you!

**Pull request merged**